kodilla-stream/src/main/java/com/kodilla/stream/StreamMain.java

```diff
@@ -1,20 +1,25 @@
 package com.kodilla.stream;
 
-import com.kodilla.stream.book.*;
+import com.kodilla.stream.forumuser.Forum;
+import com.kodilla.stream.forumuser.ForumUser;
 
+import java.time.LocalDate;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 public class StreamMain {
     public static void main(String[] args) {
 
-        BookDirectory theBookDirectory = new BookDirectory();
-        String theResultStringOfBooks = theBookDirectory.getList().stream()
-                .filter(book -> book.getYearOfPublication() > 2005)
-                .map(Book::toString)
-                .collect(Collectors.joining(",\n","<<",">>"));
+        Forum forum = new Forum();
+        Map<Integer, ForumUser> filteredUserMap = forum.getUserList().stream()
+                .filter(n -> n.getSex() == 'M')
+                .filter(n -> LocalDate.now().getYear() - n.getDateOfBirth().getYear() >= 20)
+                .filter(n -> n.getPostCount() > 0)
+                .collect(Collectors.toMap(ForumUser::getId, ForumUser -> ForumUser));
```

Comment:
przyjęta jest konwencja, że jak mapujemy obiekt na samego siebie to wykorzystuje się funkcję `Function.identity()`